Sources/CombustionBLE/DeviceManager.swift

```diff
@@ -235,9 +235,24 @@ extension DeviceManager : BleManagerDelegate {
         }
     }
     
+    func updateDeviceWithSessionInformation(identifier: UUID, sessionInformation: SessionInformation) {
+        if let probe = devices[identifier.uuidString] as? Probe {
+            probe.updateWithSessionInformation(sessionInformation)
+        }
+    }
+    
     func updateDeviceFwVersion(identifier: UUID, fwVersion: String) {
         if let device = devices[identifier.uuidString]  {
             device.firmareVersion = fwVersion
+            
+            // TODO : remove this at some point
+            // Prior to v0.8.0, the firmware did not support the Session ID command
+            // Therefore, add a hardcoded session for backwards compatibility
+            if(fwVersion.compare("v0.8.0", options: String.CompareOptions.numeric, range: nil, locale: nil) == .orderedAscending) {
```

Comment:
That `CompareOptions.numeric` type string-concatenates all numeric values in a string in order to come up with the value. This will cause issues with non-release versions of our software that were done on a branch that could potentially have numeric values in the git commit hash that's part of the firmware version (see image showing the wrong result for this comparison). 

Maybe we should do a simple `split()` here or a regex to decode the major/minor values for comparison. That would probably be a good helper function to have in this library anyway.

<img width="825" alt="image" src="https://user-images.githubusercontent.com/22039114/170540198-177204dc-7bb1-42df-b97e-7352a0896fe2.png">

Reply:
The screenshot actually shows the comparison working correctly, but it's true that the comparison can fail if the version string formats are different.  I added a split to remove debug information from version string.